src/Sentry/Threading/ScopedCountdownLock.cs

```diff
@@ -49,6 +49,11 @@ internal ScopedCountdownLock()
     /// </remarks>
     internal CounterScope TryEnterCounterScope()
     {
+        if (IsEngaged)
```

Comment:
note: the additional change of no longer allowing to `Add` new logs, while a `Flush` has been requested (in this case occurring through the `5` second timeout), which may have previously "dragged out" the Flush by still allowing new `Add` operations before the _Event_ is actually set by reaching a count of `0`